localstack-core/localstack/services/s3/provider.py

```diff
@@ -554,6 +562,7 @@ def create_bucket(
             acl=acl,
             object_ownership=request.get("ObjectOwnership"),
             object_lock_enabled_for_bucket=request.get("ObjectLockEnabledForBucket"),
+            location_constraint=location_constraint,
```

Comment:
question: in `get_bucket_location`, we'd need an empty string for `us-east-1`, but here we'd probably be storing `None`, can this be a problem? 

I think we're actually missing a proper test case for this, to validate that what AWS returns when the region is `us-east-1`: It might/should be `'<LocationConstraint xmlns="http://s3.amazonaws.com/doc/2006-03-01/"></LocationConstraint>'` and not `'<LocationConstraint xmlns="http://s3.amazonaws.com/doc/2006-03-01/">None</LocationConstraint>'` which might be the case now

Reply:
I think you're right and the attribute will probably be equal to `None` and the attribute will actually be there. So if we do that, we could distinguish between an empty string (the value was properly set) vs `None` which means the value was restored from persistence so we need to use the bucket region. 

With the `getattr` check I wrote under, we could probably differentiate between the 2 cases by checking if the value `is None`, let me know what you think :+1: